File: MYCM/support/domain/Identifiable.cs
namespace support.domain{
    /// <summary>
    /// Generic Interface for marking a domain entity identity
    /// </summary>
    /// <typeparam name="EID">Generic-Type which represents the entity identifier</typeparam>
    public interface Identifiable<EID>{
        /// <summary>
        /// Returns the entity identifier
        /// </summary>
        /// <returns>I with the entity identifier</returns>
        EID id();
        /// <summary>
        /// Checks if a certain entity identiy is the same as the current one
        /// </summary>
        /// <param name="comparingEntity">I with the entity identity being compared to the current one</param>
        /// <returns>bool if both entities identities are the same</returns>
        //#TODO See Version that .NET is using for C#, default implementations are available since C# 8
        bool sameAs(EID comparingEntity);
    }
}

File: MYCA/myca/support/src/main/java/cdiomyc/support/domain/Identifiable.java
package cdiomyc.support.domain;

/**
 * Generic interface for identifying domain entities
 * @author <a href="https://github.com/freitzzz">freitzzz</a>
 * @param <EID> Generic-Type of the domain entity identifier
 */
public interface Identifiable<EID> {
    /**
     * Returns the domain entity identifier
     * @return EID with the domain entity identifier
     */
    EID id();
    /**
     * Checks if a comparing domain entity identifier is the same as 
     * the current domain entity identifier
     * @param comparingIdentifier EID with the comparing domain entity identifier
     * @return boolean true if both identifiers are equal, false if not
     */
    default boolean sameAs(EID comparingIdentifier){return id().equals(comparingIdentifier);};
}
